iceoryx_hoofs/testing/include/iceoryx_hoofs/testing/mocks/error_handler_mock.hpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_TESTUTILS_ERROR_HANDLER_MOCK_HPP
+#define IOX_HOOFS_TESTUTILS_ERROR_HANDLER_MOCK_HPP
+
+#include "iceoryx_hoofs/error_handling/error_handling.hpp"
+#include <mutex>
+
+#include "iceoryx_hoofs/testing/test.hpp"
+
+namespace iox
+{
+template <typename Error>
+using TypedHandlerFunction = std::function<void(const Error, const ErrorLevel)>;
+
+class ErrorHandlerMock : protected ErrorHandler
+{
+  public:
+    template <typename Error>
+    static cxx::GenericRAII setTemporaryErrorHandler(const TypedHandlerFunction<Error>& newHandler) noexcept;
+
+  protected:
+    /// Needed, if you want to exchange the handler. Remember the old one and call it if it is not your error. The error
+    /// mock needs to be the last one exchanging the handler in tests.
+    static std::mutex handler_mutex;
+};
+
+template <typename Error>
+cxx::optional<iox::TypedHandlerFunction<Error>> typedHandler;
+
+template <typename ErrorEnumType>
+inline void errorHandlerForTest(const uint32_t error, const char* errorName, const ErrorLevel level) noexcept
+{
+    uint32_t errorEnumType = error >> 16;
```

Comment:
more like `errorCode`, certainly no type (I know it means probably type of error, but I think it is a misleading name). I would drop `Enum` in any case.

Reply:
I would appreciate to leave the naming as it is, if you're ok with it. Reason is that this makes it easier to understand and/or remember what is happening here and how the type of error enum is detected.

Reply:
How about `errorDomain`/`errorEnumDomain`?

Reply:
When I heard domain, in this context I think of a tld like `bar.com`. Happy to discuss the naming, but for now I would appreciate if we could merge it as it is.